src/clp_logging/handlers.py

```
@@ -996,3 +1012,276 @@ def _serialize_kv_pair_log_event(
+                    Key=self._obj_key,
+                    Body=data,
+                    ContentEncoding='zstd' if self.enable_compression else 'binary',
+                    ChecksumSHA256=sha256_checksum
+                )
+
+                # Verify integrity of the upload using SHA256 Checksum
+                response = self.s3_client.head_object(Bucket=self.s3_bucket, Key=self.obj_key)
+                if 'ChecksumSHA256' in response:
+                    s3_checksum = response['ChecksumSHA256']
+                    if s3_checksum != sha256_checksum:
+                        raise ValueError(f"Checksum mismatch. Upload aborted.")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)